envoy/http/async_client.h

```diff
@@ -379,11 +379,14 @@ class AsyncClient {
       sampled_ = sampled;
       return *this;
     }
-
     StreamOptions& setSidestreamWatermarkCallbacks(SidestreamWatermarkCallbacks* callbacks) {
       sidestream_watermark_callbacks = callbacks;
       return *this;
     }
+    StreamOptions& setMatchingRoute(Envoy::Router::RouteConstSharedPtr route) {
+      matching_route = route;
+      return *this;
+    }
```

Comment:
I think the ParentContext will contains a stream info pointer and the stream info contains pointer to matched route? So, this should be unnecessary? You can reuse the `setParentContext`.

Reply:
/wait